clusterloader2/pkg/measurement/common/slos/pod_startup_latency.go

```diff
@@ -146,6 +148,29 @@ func (p *podStartupLatencyMeasurement) start(c clientset.Interface) error {
 		p.addEvent,
 	)
 	go p.processEvents()
+
+	selector := fields.Set{
+		"involvedObject.kind": "Pod",
+		"source":              schedulerName,
+	}.AsSelector().String()
+
+	p.stopSchedCh = make(chan struct{})
+
+	e := informer.NewInformer(
```

Comment:
Could we use [`Controller`](https://github.com/kubernetes/client-go/blob/cbe28cf468c43969d2c170e079f556becfc7adb7/tools/cache/controller.go#L116-L123), a slightly lower-lever primitive than `Informer` and pass a large (e.g. 10k) `WatchListPageSize` in the [Config](https://github.com/kubernetes/client-go/blob/cbe28cf468c43969d2c170e079f556becfc7adb7/tools/cache/controller.go#L39-L79)?

The reason I'm asking this is that in large clusters there's a tendency to have O(hundreds of thousands) events and listing them using default page size ([500](https://github.com/kubernetes/client-go/blob/ec0f33729df006747d8685920630bacb364bf3bc/tools/pager/pager.go#L31)) may result in informer's initial list getting timed out.

Reply:
>listing them using default page size ([500](https://github.com/kubernetes/client-go/blob/ec0f33729df006747d8685920630bacb364bf3bc/tools/pager/pager.go#L31)) may result in informer's initial list getting timed out.

Oh! how will it timeout  IIUC ? I have ensured we are not relying on client side timeout defined [here](https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/measurement/util/informer/informer.go#L84-L91) for this use-case,  instead I'm calling directly  `Run` method here [here](https://github.com/kubernetes/perf-tests/blob/b7234715ccb6626f7e9b92cba5e89204e083728b/clusterloader2/pkg/measurement/common/slos/pod_startup_latency.go#L173) . Am i misinterpreting what you are trying to imply here ?

Reply:
Ah, I see... we deliberately set the timeout in our wrappers around informers to make sure that the initial list (the one responsible for the cache's sync) completes in a reasonable time. In a clusters with O(xxx k) events this will take ages (O(a few minutes)) if using the default page size and because of that possibly run into the "too old resource version" during the initial list. I'd strongly suggest using the larger page size and hence the `Controller` primitive for the List+Watch pattern for events like we do in https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/measurement/common/loadbalancer_nodesync_latency.go.

CC @mborsz for his thoughts as I'll be OOO for the rest of the week.